File: kotlin/searching/LinearSearch.kt
fun linearSearch(nums: IntArray, target: Int): Int { // Time Complexity = O(n)
    for (i in nums.indices) {
        if (nums[i] == target) {
            return i
        }
    }
    return -1
}

fun main() {
    val nums = intArrayOf(1, 2, 3, 4, 5, 6, 7, 8, 9)
    val target = 6
    println(linearSearch(nums, target))
}

File: c++/searching/LinearSearch.cpp
#include <iostream>
#include <vector>

using namespace std;

int linearSearch(vector<int> nums, int target){ // Time Complexity = O(n)
    for (int i = 0; i < nums.size(); i++)
    {
        if (nums[i] == target)
        {
            return i;
        }
    }
    return -1;
}

int main(){
    vector<int> nums = {1,2,3,4,5,6,7,8,9};
    int target = 6;
    cout << linearSearch(nums, target) << "\n";
    return 0;
}